### DISCUSSION TITLE
How to use UDPStream with UIPEthernet?

### DISCUSSION THREAD

**Discussion by leelx78** [pschatzmann/arduino-audio-tools#1061] [Q&A] [answered]
Hi @pschatzmann hope you are ok.

I have an application where I need to use Ethernet (UIPEthernet Library, see https://github.com/UIPEthernet/UIPEthernet) instead of WiFi so I headed to your `UDPStream : public WiFiUDP` class but didn't see an obvious way how to get it to work with this library.

My socket is initialised as `EthernetUDP udpSocket` and in principle it should have already most of the stream methods required but before reinventing the wheel and starting over by extending `EthernetUDP` to create a compatible `AudioStream' I thought to ask you for some advice. How do you suggest this should be done most efficiently?

Also, do you plan to add support for Ethernet shields in the future?

Thanks!

**Comment by pschatzmann** [comment, marked as answer]
I was lazy when I implemented that class. Can you double check if it would work if I change it to the [following more generic implementation](https://gist.github.com/pschatzmann/8903960ec39e31f70f0a8aa486180577):

It should work to pass a EthernetUDP object in the constructor...

**Comment by pschatzmann** [comment]
Did you have the chance to check if the proposed changes are working ?
I will only consider to change my code after a positive feed-back...

**Reply by leelx78** [reply]
Hi Phil, sorry for the delay, I'm stuck on another project running in parallel. I'll update you on this as soon as I have a chance.

Thanks!

**Comment by leelx78** [comment]
Hi @pschatzmann just tested the updated and works great. I only needed to add a method 
```
    void setUDPClass(UDP &udp)
    {
      p_udp = &udp;
    };
```
to pass the UDP stream after the class is created. Can you include this method as well?

Thanks and congrats again for such an incredible library!

**Comment by pschatzmann** [comment]
I decided to rename the method  void setUDP(UDP &udp)
